aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/events/ModelSyncedEvent.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.datastore.syncengine.events;
+
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.datastore.syncengine.SyncType;
+
+/**
+ * Hub event payload emitted when the initial sync completes for a given model.
+ */
+public final class ModelSyncedEvent {
```

Comment:
Since you'll send these back to the customer, be sure that all of your event structures are:
1. Immutable;
2. Implement equals, toString, and hashCode
3. If applicable, implementing `Comparable` might be appropriate as well?

Reply:
1 and 2 should be addressed. I'm not sure it makes sense to implement comparable here since none of the fields really stand out in terms of sorting.